qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteredSnapshotTest.java

```diff
@@ -131,34 +132,127 @@ public void shouldIncludeExportedPositionInSnapshot() {
     publishMessages();
 
     // when - then
+    awaitUntilAsserted(
+        (broker) -> {
+          triggerSnapshotRoutine();
+          assertThat(broker)
+              .havingSnapshot()
+              .withExportedPosition(ControllableExporter.lastUpdatedPosition);
+        });
+  }
+
+  @Test
+  public void shouldTakeSnapshotWhenExporterPositionIsMinusOne() {
+    // given
+    // an exporter is configured, but nothing gets exported
+    ControllableExporter.updatePosition(false);
+    publishMessages();
+
+    // when
+    triggerSnapshotRoutine();
+
+    // then
+    awaitUntilAsserted(
+        (broker) -> {
+          assertThat(broker).havingSnapshot().withIndex(0).withTerm(0).withExportedPosition(0);
+        });
+  }
+
+  @Test
+  public void shouldKeepIndexAndTerm() {
+    // given
+    ControllableExporter.updatePosition(false);
+    removeExporters();
+    restartCluster();
+    publishMessages();
+    triggerSnapshotRoutine();
+
+    // expect - each broker has created a snapshot
+    awaitUntilAsserted(
+        (broker) -> {
+          assertThat(broker).havingSnapshot().withExportedPosition(Long.MAX_VALUE);
+        });
+
+    final Map<Integer, SnapshotId> snapshotsByBroker =
+        clusteringRule.getTopologyFromClient().getBrokers().stream()
+            .collect(Collectors.toMap(BrokerInfo::getNodeId, this::getSnapshot));
+
+    // when
+    configureExporters();
+    restartCluster();
+    publishMessages();
+    triggerSnapshotRoutine();
+
+    // then
+    awaitUntilAsserted(
+        (broker) -> {
+          final SnapshotId expectedSnapshot =
+              snapshotsByBroker.get(broker.getConfig().getCluster().getNodeId());
+          assertThat(broker)
+              .havingSnapshot()
+              .withIndex(expectedSnapshot.getIndex())
+              .withTerm(expectedSnapshot.getTerm());
+        });
+  }
+
+  @Test
+  public void shouldNotTakeNewSnapshot() {
+    // given
+    ControllableExporter.updatePosition(false);
+    removeExporters();
+    restartCluster();
+    publishMessages();
+
+    awaitUntilAsserted(
+        (broker) -> {
+          triggerSnapshotRoutine();
+          clusteringRule.getBrokers().stream()
+              .filter(b -> !b.equals(broker))
+              .forEach(
+                  (b) -> {
+                    final SnapshotId snapshot = clusteringRule.getSnapshot(b).get();
+                    final long expectedProcessedPosition = snapshot.getProcessedPosition();
+                    assertThat(broker)
+                        .havingSnapshot()
+                        .withProcessedPosition(expectedProcessedPosition);
+                  });
+        });
+
+    // expect - each broker has created a snapshot with exported position = Long.MAX_VALUE
+    awaitUntilAsserted(
+        (broker) -> {
+          assertThat(broker).havingSnapshot().withExportedPosition(Long.MAX_VALUE);
+        });
```

Comment:
:wrench: May be the following is more compact and readable. 

```suggestion
    final var leaderId = clusteringRule.getLeaderForPartition(1).getNodeId();
    final var leaderAdminService =
        clusteringRule.getBroker(leaderId).getBrokerContext().getBrokerAdminService();
    final var expectedProcessedPosition =
        leaderAdminService.getPartitionStatus().get(1).getProcessedPosition();

    awaitUntilAsserted(
        broker -> {
          triggerSnapshotRoutine();
          assertThat(broker)
              .havingSnapshot()
              .withProcessedPosition(expectedProcessedPosition)
              .withExportedPosition(Long.MAX_VALUE);
        });
```

Reply:
👍